utils/oomparser/oomparser.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oomparser
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/cadvisor/utils"
+)
+
+var containerRegexp *regexp.Regexp = regexp.MustCompile(
+	`Task in (.*) killed as a result of limit of `)
+var lastLineRegexp *regexp.Regexp = regexp.MustCompile(
+	`(^[A-Z]{1}[a-z]{2} [0-9]{1,2} [0-9]{1,2}:[0-9]{2}:[0-9]{2}) .* Killed process ([0-9]+) \(([0-9A-Za-z_]+)\)`)
+var firstLineRegexp *regexp.Regexp = regexp.MustCompile(
+	`kernel: [[0-9]+.[0-9]+] [0-9A-Za-z_]+ invoked oom-killer:`)
+
+// struct to hold a list of OomInstance objects
+type OomParser struct {
+	oomInstances []*OomInstance
+	systemFile   string
+}
+
+// struct that contains information related to an OOM kill instance
+type OomInstance struct {
+	// process id of the killed process
+	Pid int
+	// the name of the killed process
+	ProcessName string
+	// the time that the process was reported to be killed,
+	// accurate to the minute
+	TimeOfDeath time.Time
+	// the absolute name of the container that OOMed
+	ContainerName string
+}
+
+type SliceOomInstance []*OomInstance
+
+func (self SliceOomInstance) String() string {
+	returnstring := ""
+	for i, currentOom := range self {
+		returnstring += fmt.Sprintf(
+			"oomKill instance %d. %v\n", i,
+			currentOom)
+	}
+	return returnstring
+}
+
+// public function to retrieve list of OomInstances in the oomLog struct
+func (self OomParser) GetAllOoms() []*OomInstance {
+	return self.oomInstances
+}
+
+// gets the container name from a line and adds it to the oomInstance.
+func getContainerName(line string, currentOomInstance *OomInstance) error {
+	parsedLine := containerRegexp.FindStringSubmatch(line)
+	if parsedLine == nil {
+		return nil
+	}
+	currentOomInstance.ContainerName = parsedLine[1]
```

Comment:
Oh, one final thing: make this: path.Join("/", parsedLine[1]) so that it is an absolute container name.